### PR TITLE
[C-2689] Add upload confirmation modal

### PR DESCRIPTION
**Pull request by Kyle-Shanks**
### Description
Add upload confirmation modal

### Dragons

N/A

### How Has This Been Tested?

Manually tested

### How will this change be monitored?

N/A

### Feature Flags ###

N/A

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/kj-Add-upload-confirmation-modal

File: packages/web/src/components/upload-confirmation-modal/UploadConfirmationModal.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    'Ready to begin uploading? Your followers will be notified once your upload is complete.',
+  hiddenDescription: 'Ready to begin uploading?',
+  cancel: 'Go Back',
+  upload: 'Upload'
+}
+
+export const UploadConfirmationModal = () => {
+  const confirmCallback = useSelector(getConfirmCallback)
+  const hasPublicTracks = useSelector(getHasPublicTracks)
+  const [isOpen, setIsOpen] = useModalState('UploadConfirmation')
```
Comment (amendelsohn) [inline packages/web/src/components/upload-confirmation-modal/UploadConfirmationModal.tsx:37]: should we make the modal names constants somewhere?

File: packages/web/src/components/upload-confirmation-modal/UploadConfirmationModal.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  return (
+    <Modal isOpen={isOpen} onClose={onClose} size='small'>
+      <ModalHeader>
+        <ModalTitle
+          icon={<IconUpload className={styles.titleIcon} />}
+          title={
+            <Text
+              variant='label'
+              size='xLarge'
+              strength='strong'
```
Comment (amendelsohn) [inline packages/web/src/components/upload-confirmation-modal/UploadConfirmationModal.tsx:57]: 💪

File: packages/web/src/pages/upload-page/UploadPageNew.tsx [hunk 3]
```diff
@@ -99,7 +105,11 @@ export const UploadPageNew = () => {
             formState={formState}
             onContinue={(formState: UploadFormState) => {
               setFormState(formState)
-              setPhase(Phase.FINISH)
+              const hasPublicTracks =
+                formState.tracks?.some(
+                  (track) => !track.metadata.is_unlisted
+                ) ?? true
```
Comment (amendelsohn) [inline packages/web/src/pages/upload-page/UploadPageNew.tsx:111]: if there are no tracks one could argue that there are no public tracks? 🤷
though I can see just keeping the default true to make sure we don't miss it

**Review by amendelsohn — APPROVED** [review]
15 files for 1 little modal 😵
Marcus, save us!

File: packages/web/src/pages/upload-page/UploadPageNew.tsx (hunk 3 above)
Comment (Kyle-Shanks) [reply packages/web/src/pages/upload-page/UploadPageNew.tsx:111]: I dont think you could even see this without having tracks in the form state, but I figured the public message would be a fine default 👍

Comment (amendelsohn) [reply packages/web/src/pages/upload-page/UploadPageNew.tsx:111]: yea sounds good

File: packages/web/src/components/upload-confirmation-modal/UploadConfirmationModal.tsx (hunk 1 above)
Comment (Kyle-Shanks) [reply packages/web/src/components/upload-confirmation-modal/UploadConfirmationModal.tsx:37]: They are defined in the Modals type and then auto completed everywhere else

Comment (amendelsohn) [reply packages/web/src/components/upload-confirmation-modal/UploadConfirmationModal.tsx:37]: Ohh neat!

**Pull request merged**